### PR TITLE
Backmerge: Projectstructure, Linters, Configuration, Package.json

### PR DESCRIPTION
**Pull request by andre-meyer**
### Backmerge (Redo) 01

Includes changes for:
- Entrypoint `index.js`
- Store Configuration `store.js`
- General Projectstructure
- Linting Configuration
- `package.json`
- Configuration Files

### CODE REVIEW

File: nginx.template [hunk 1]
```diff
@@ -18,6 +18,14 @@ server {
 
   keepalive_timeout   70;
   add_header          Alternate-Protocol  $NGINX_SSL_PORT:npn-spdy/2;
+
+  location ~* ^.+.(jpg|jpeg|png|gif|ico|svg|css|js)$ {
+    expires 30d;
```
Comment (apanizo) [inline nginx.template]: What is this? 30d cache for js files?

File: package.json [hunk 2]
```diff
@@ -48,6 +52,7 @@
     "normalizr": "^3.2.2",
     "prop-types": "^15.6.0",
     "raven-js": "^3.21.0",
+    "raven-for-redux": "^1.1.1",
```
Comment (apanizo) [inline package.json]: What does it do?

File: src/config.json [hunk 3]
```diff
@@ -17,5 +20,6 @@
     "host": "rinkeby.infura.io",
     "port": 443
   },
-  "fetchMarketTimeInterval": 15000
+  "fetchMarketTimeInterval": 15000,
+  "ravenPublicDSN": null
```
Comment (apanizo) [inline src/config.json]: What is this?

File: webpack.dev.config.js [hunk 4]
```diff
@@ -106,18 +102,25 @@ module.exports = {
...
     }),
-    new webpack.DefinePlugin({
-      'process.env': {
-        VERSION: JSON.stringify(`${version}#${build}`),
-        NODE_ENV: JSON.stringify(nodeEnv),
-        GNOSISDB_URL: JSON.stringify(gnosisDbUrl),
-        ETHEREUM_URL: JSON.stringify(ethereumUrl),
-        WHITELIST: JSON.stringify(whitelist),
-      },
+    new webpack.EnvironmentPlugin({
```
Comment (apanizo) [inline webpack.dev.config.js]: Are you sure we want to use ```EnvironmentPlugin```? IF my memory does not fail we switched to DefinePlugin, right?

I specifically remember during the last deployment Giacomo told me: "Use ```WHITELIST: JSON.stringify(whitelist),```.

What am I missing?

File: webpack.prod.config.js [hunk 5]
```diff
@@ -94,18 +93,25 @@ module.exports = {
...
-    new webpack.DefinePlugin({
-      'process.env': {
-        VERSION: JSON.stringify(`${version}#${build}`),
-        NODE_ENV: JSON.stringify(nodeEnv),
-        GNOSISDB_URL: JSON.stringify(gnosisDbUrl),
-        ETHEREUM_URL: JSON.stringify(ethereumUrl),
-        WHITELIST: JSON.strinfigy(whitelist),
-      },
+    new webpack.EnvironmentPlugin({
+      VERSION: `${version}#${build}`,
```
Comment (apanizo) [inline webpack.prod.config.js]: Same here?

File: nginx.template (hunk 1 above)
Comment (denisgranha) [reply nginx.template]: we don't use nginx anymore for our servers, so this file is not relevant

File: package.json (hunk 2 above)
Comment (andre-meyer) [reply package.json]: it's for sentry, I'm already adding it

File: src/config.json (hunk 3 above)
Comment (andre-meyer) [reply src/config.json]: Sentry.io, already added the key

File: webpack.dev.config.js (hunk 4 above)
Comment (andre-meyer) [reply webpack.dev.config.js]: Nope, we use `EnvironmentPlugin` on both `webpack.prod.js` and `webpack.dev.js`. Doesn't require to use stringify

File: webpack.prod.config.js (hunk 5 above)
Comment (andre-meyer) [reply webpack.prod.config.js]: Nope nope, Environment Plugin just like in Olympia

File: webpack.dev.config.js (hunk 4 above)
Comment (apanizo) [reply webpack.dev.config.js]: @denisgranha @giacomolicari ? Will we have any problem?

Comment (denisgranha) [reply webpack.dev.config.js]: Don't remember the scenario but the whitelisted accounts weren't being used until we added stringify.
Test your solution with both enviroments and whatever is better
I'm not an expert about those plugins
It's up to you

File: nginx.template (hunk 1 above)
Comment (andre-meyer) [reply nginx.template]: Will remove it

File: webpack.dev.config.js (hunk 4 above)
Comment (andre-meyer) [reply webpack.dev.config.js]: https://webpack.js.org/plugins/environment-plugin/#usage

This plugin automatically `stringify`s all values :)

Comment (andre-meyer) [reply webpack.dev.config.js]: I had Mikhail test it for olympia, it's in both configurations now and the last problem we had, had nothing to do with this

**Review by apanizo — APPROVED** [review]

**Pull request merged**